lisa/analysis/frequency.py

```diff
@@ -569,4 +573,34 @@ def plot_domain_frequency_transitions(self, pct: bool=False):
             )
         ).cols(1)
 
+    @TraceAnalysisBase.plot_method
+    @df_peripheral_clock_effective_rate.used_events
+    def plot_peripheral_frequency(self, clk_name, average: bool=True):
+        """
+        Plot frequency for the specified peripheral clock frequency
+
+        :param clk_name: The clock name for which to plot frequency
+        :type clk_name: str
+
+        :param average: If ``True``, add a horizontal line which is the
+            frequency average.
+        :type average: bool
+
+        """
+        logger = self.logger
+        df = self.df_peripheral_clock_effective_rate(clk_name)
+
+        freq = series_refit_index(df['effective_rate'], window=self.trace.window)
+        avg = series_mean(freq)
+        logger.info(
+            "Average frequency for {} : {:.3f} GHz".format(clk_name, avg / 1e6))
+
+        df = df_refit_index(df, window=self.trace.window)
+        fig = plot_signal(df['effective_rate'], name=f'Frequency of {clk_name} (Hz)')
+
+        if average and avg > 0:
```

Comment:
`avg > 0` can be simplified into `avg`, in case it becomes negative for some reason (would probably be a bug but at least it would make it visible)